signer/src/transaction_signer.rs

```diff
@@ -425,6 +535,125 @@ where
         Ok(())
     }
 
+    fn create_new_state_machine(&mut self) -> Result<SignerStateMachine, error::Error> {
+        let signers: hashbrown::HashMap<u32, _> = self
+            .signer_public_keys
+            .iter()
+            .cloned()
+            .enumerate()
+            .map(|(id, key)| {
+                id.try_into()
+                    .map(|id| (id, key))
+                    .map_err(|_| error::Error::TypeConversion)
+            })
+            .collect::<Result<_, _>>()?;
+        let key_ids = signers
+            .clone()
+            .into_iter()
+            .map(|(id, key)| (id + 1, key))
+            .collect();
+
+        let public_keys = wsts::state_machine::PublicKeys { signers, key_ids };
+
+        let threshold = self.threshold;
+        let num_parties = self
+            .signer_public_keys
+            .len()
+            .try_into()
+            .map_err(|_| error::Error::TypeConversion)?;
+        let num_keys = num_parties;
+
+        let signer_pub_key = self.signer_pub_key()?;
+
+        let id: u32 = self
+            .signer_public_keys
+            .iter()
+            .enumerate()
+            .find(|(_, key)| *key == &signer_pub_key)
+            .ok_or_else(|| error::Error::MissingPublicKey)?
+            .0
+            .try_into()
+            .map_err(|_| error::Error::TypeConversion)?;
+
+        let key_ids = vec![id + 1];
+
+        let state_machine = SignerStateMachine::new(
+            threshold,
+            num_parties,
+            num_keys,
+            id,
+            key_ids,
+            self.signer_private_key,
+            public_keys,
+        );
+
+        Ok(state_machine)
+    }
+
+    async fn creat_state_machine_with_loaded_state(
```

Comment:
I think we should probably add some documentation to this function. It seems pretty important, so a summary here will go along way. Some important parts that

Reply:
Makes sense, I've added a docstring now. Let me know if there's anything you'd like me to elaborate on.